blog/content/post/2024-11-26-measured-boot-improvements.md

```diff
@@ -0,0 +1,222 @@
+reproducible PCR2
+value](https://github.com/Dasharo/dasharo-issues/issues/455).
+
+## Summary
+
+TPM event log integrity is an excellent achievement for our Dasharo team. While we
+did an enormous work to improve it, there are still some aspects not covered
+by current implementation:
+
+1. What if the PCR banks are changed?
+
+   The EDKII TPM2 setup menu allows changing PCR banks, effectively
+   influencing what measurements (precisely hash algorithms used to measure
+   firmware components) may be recorded in the TPM. coreboot is hardcoded to
+   use only one hashing algorithm (fixed at build). It may result in PCR banks
+   not being properly extended or the PCR values being entirely out of sync
+   with the TPM event log. coreboot should query the TPM for currently enabled
+   PCR banks and use all algorithms enabled at the time. That, of course, may
+   have various implications:
+
+    - Pre-RAM measurements are saved in Cache-as-RAM (CAR) memory, which is
+      limited. Calculating more hashes will cause the CAR memory to shrink,
+      which effectively limits the amount of cache for silicon initialization
+      code. It may not be a big deal for modern platforms with big caches, but
+      rather for the old ones.
+    - coreboot supports only one algorithm in a single event log entry.
+      Support for an agile format would have to be implemented.
+
+2. What about pre-reset vector measurements like Boot Guard measured boot?
+
+   Intel Boot Guard is a hardware Root of Trust feature that allows the
+   firmware to be verified (and optionally measured) before the CPU is
+   released from reset. It guarantees the integrity and reliable measurement
+   of the critical early boot code. Currently, coreboot ignores the fact that
+   something may have extended TPM PCRs with measurements before coreboot
+   initialized the TPM or started logging measurements. Support for Boot
+   Guard-created measurements must also be added to keep the PCR values
+   consistent with the TPM event log.
+
+Improving the security of the firmware is a process. What is important is to
+take small steps and constantly move forward by improving things, one after
+another. So, let's collaborate and work together on the safety of our systems.
+Unlock the full potential of your hardware and secure your firmware with the
+experts at 3mdeb! If you're looking to boost your product's performance and
+protect it from potential security threats, our team is here to help.
+[Schedule a call with
+us](https://calendly.com/3mdeb/consulting-remote-meeting) or drop us an email
+at `contact<at>3mdeb<dot>com` to start unlocking the hidden benefits of your
+hardware. And if you want to stay up-to-date on all things firmware security
+and optimization, be sure to [sign up for our
+newsletter](https://newsletter.3mdeb.com/subscription/PW6XnCeK6). Don't let
+your hardware hold you back, work with 3mdeb to achieve more!
```

Comment:
I would appreciate adding in between the lines pitch about training related to security and trustworthiness:
https://3mdeb.com/training/
As well as a training pitch about the upcoming HWIO25 USA: https://hardwear.io/usa-2025/training/mastering-uefi-secure-boot.php

Reply:
fac86ce1750d8cbe5b12e66b34e7208a234b786e